gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+        TeV_units = ['1/s/TeV/sr', 's-1 sr-1 TeV-1', '1 / (s sr TeV)',
+                     '1 / (TeV s sr)']
+        MeV_units = ['1/s/MeV/sr', 'MeV-1 s-1 sr-1', '1 / (s sr MeV)',
+                     '1 / (MeV s sr)']
+        if background_unit in TeV_units:
+            background_unit = '1 / (s TeV sr)'
+        elif background_unit in MeV_units:
+            background_unit = '1 / (s MeV sr)'
+        else:
+            raise ValueError("Cannot interpret units ({})".format(background_unit))
+        background = Quantity(background, background_unit)
+
+        return CubeBackgroundModel(detx_bins=detx_bins,
+                                   dety_bins=dety_bins,
+                                   energy_bins=energy_bins,
+                                   background=background)
+
+    @staticmethod
+    def from_fits_image(tbhdu):
+        """Read cube background model from a fits image.
+
+        Parameters
+        ----------
+        tbhdu : `~astropy.io.fits.BinTableHDU`
+            HDU binary table for the bg cube
+
+        Returns
+        -------
+        bg_cube : `~gammapy.models.CubeBackgroundModel`
+            bg model cube object
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def read(filename, format='bin_table'):
+        """Read cube background model from fits file.
+
+        Several input formats are accepted, depending on the value
+        of the 'format' parameter:
+
+        * bin_table (default and preferred format): `~astropy.io.fits.BinTableHDU`
+        * image (alternative format): `~astropy.io.fits.ImageHDU`
+
+        Parameters
+        ----------
+        filename : `~string`
+            name of file with the bg cube
+        format : `~string`, optional
+            format of the bg cube to read
+
+        Returns
+        -------
+        bg_cube : `~gammapy.models.CubeBackgroundModel`
+            bg model cube object
+        """
+        hdu = fits.open(filename)
+        if format == 'bin_table':
+            return CubeBackgroundModel.from_fits_bin_table(hdu)
+        elif format == 'image':
+            return CubeBackgroundModel.from_fits_image(hdu)
+        else:
+            raise ValueError("Got format {}, which is not understood."
+                             .format(format))
+
+    def to_astropy_table(self):
+        """Convert cube background model to astropy table format.
+
+        The name of the table is stored in the table meta information
+        under the keyword 'name'.
+
+        Returns
+        -------
+        table : `~astropy..table.Table`
+            table containing the bg cube
+        """
+        # fits unit string
+        u_detx = '{0.unit:FITS}'.format(self.detx_bins)
+        u_dety = '{0.unit:FITS}'.format(self.dety_bins)
+        u_energy = '{0.unit:FITS}'.format(self.energy_bins)
+        u_bg = '{0.unit:FITS}'.format(self.background)
+
+        # data arrays
+        a_detx_lo = Quantity([self.detx_bins[:-1]])
```

Comment:
Why call Quantity here? Aren't these quantity objects already (they should be I think)?
I.e. these lines can go and you can just assign data members from `self` to table columns below?